### ISSUE TITLE
input_params variable not available in jinja2 context

### ISSUE THREAD

**Issue by deepbrook** [kapicorp/kapitan#789]
I'm trying to compile a jinja template with the following compile job:

```
parameters:
  kapitan:
    compile:
      - input_type: jinja2
        input_params:
          app: "test-app"
          instance: "test-app-42"
```

The documentation on the `input_params` is a bit thin, but after inspecting the python code I was under the impression
the value of `input_params` is attached to the context passed to the [render_jinja_file](https://github.com/kapicorp/kapitan/blob/1cd429102dda0a7cee6525a2e61c2b88f50ab292/kapitan/utils.py#L126) function.

However, when I try to reference `input_params` in my jinja2 template like this:

```
---
apiVersion: v1
kind: Deployment
metadata:
  name:  "{{ input_params.app }}"
  labels: &labels
    app: {{ input_params.app }}
    managed-by: kubectl
```

.. I get an `undefined` variable:

```
2021-10-29 16:57:34,832 kapitan.utils DEBUG    render_jinja2 rendering /home/deepbrook/devel/systems/streaming/vod-msk-cluster/deploy/components/k8s/deployment.yaml.j2
2021-10-29 16:57:34,843 kapitan.targets DEBUG    Compile pool terminated
2021-10-29 16:57:34,843 kapitan.targets ERROR

2021-10-29 16:57:34,843 kapitan.targets ERROR    Jinja2 error: failed to render /home/deepbrook/devel/systems/streaming/vod-msk-cluster/deploy/components/k8s/deployment.yaml.j2: 'input_params' is undefined
Compile error: failed to compile target: dev
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/home/deepbrook/.local/lib/python3.8/site-packages/kapitan/utils.py", line 164, in render_jinja2
    "content": render_jinja2_file(
  File "/home/deepbrook/.local/lib/python3.8/site-packages/kapitan/utils.py", line 132, in render_jinja2_file
    return env.get_template(filename).render(context)
  File "/home/deepbrook/.local/lib/python3.8/site-packages/jinja2/environment.py", line 1304, in render
    self.environment.handle_exception()
  File "/home/deepbrook/.local/lib/python3.8/site-packages/jinja2/environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/home/deepbrook/devel/systems/streaming/vod-msk-cluster/deploy/components/k8s/deployment.yaml.j2", line 5, in top-level template code
    name:  "{{ inventory.parameters.common.resource_prefix }}.{{ input_params.app }}"
  File "/home/deepbrook/.local/lib/python3.8/site-packages/jinja2/environment.py", line 474, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'input_params' is undefined

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/deepbrook/.local/lib/python3.8/site-packages/kapitan/inputs/base.py", line 69, in compile_input_path
    self.compile_file(
  File "/home/deepbrook/.local/lib/python3.8/site-packages/kapitan/inputs/jinja2.py", line 39, in compile_file
    for item_key, item_value in render_jinja2(
  File "/home/deepbrook/.local/lib/python3.8/site-packages/kapitan/utils.py", line 170, in render_jinja2
    raise CompileError(f"Jinja2 error: failed to render {render_path}: {e}")
kapitan.errors.CompileError: Jinja2 error: failed to render /home/deepbrook/devel/systems/streaming/vod-msk-cluster/deploy/components/k8s/deployment.yaml.j2: 'input_params' is undefined

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/home/deepbrook/.local/lib/python3.8/site-packages/kapitan/targets.py", line 465, in compile_target
    input_compiler.compile_obj(comp_obj, ext_vars, **kwargs)
  File "/home/deepbrook/.local/lib/python3.8/site-packages/kapitan/inputs/base.py", line 54, in compile_obj
    self.compile_input_path(input_path, comp_obj, ext_vars, **kwargs)
  File "/home/deepbrook/.local/lib/python3.8/site-packages/kapitan/inputs/base.py", line 79, in compile_input_path
    raise CompileError("{}\nCompile error: failed to compile target: {}".format(e, target_name))
kapitan.errors.CompileError: Jinja2 error: failed to render /home/deepbrook/devel/systems/streaming/vod-msk-cluster/deploy/components/k8s/deployment.yaml.j2: 'input_params' is undefined
Compile error: failed to compile target: dev
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/deepbrook/.local/lib/python3.8/site-packages/kapitan/targets.py", line 136, in compile_targets
    [p.get() for p in pool.imap_unordered(worker, target_objs) if p]
  File "/home/deepbrook/.local/lib/python3.8/site-packages/kapitan/targets.py", line 136, in <listcomp>
    [p.get() for p in pool.imap_unordered(worker, target_objs) if p]
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 868, in next
    raise value
kapitan.errors.CompileError: Jinja2 error: failed to render /home/deepbrook/devel/systems/streaming/vod-msk-cluster/deploy/components/k8s/deployment.yaml.j2: 'input_params' is undefined
Compile error: failed to compile target: dev
2021-10-29 16:57:34,845 kapitan.targets DEBUG    Removed /tmp/tmpd7yxsw0s.kapitan
```

How do I access values I defined in `input_params` ?

**Comment by ramaro** [comment]
Hey @deepbrook thanks for reporting. I'm assuming you're running `0.29.5` which has a multiprocessing race condition when accessing the inventory. This has been fixed in [0.30.0-rc0](https://github.com/kapicorp/kapitan/releases/tag/v0.30.0-rc.0) and you should  be able to validate by compiling the ` jinja2-input-params` target in `kapitan/tests/test_resources`.
Let me know if that works for you.

**Comment by deepbrook** [comment]
@ramaro I was not aware of that! I've upgraded as you suggested and all works as expected. :)
